logstash-core/lib/logstash/instrument/metric_type/base.rb

```diff
@@ -0,0 +1,16 @@
+# encoding: utf-8
+require "logstash/instrument/metric_type/counter"
+
+module LogStash module Instrument module MetricType
+  class Base
+    def initialize(namespaces, key)
+      @namespaces = namespaces
+      @key = key
+    end
+
+    private
+    def type
+      @type ||= self.class.name.split("::").last.downcase
```

Comment:
I don't create a class for the name anymore, see https://github.com/elastic/logstash/pull/4358/files#diff-e456278bc07b599d01b2c59ec369bbcdR6


I could move the `self.class.name.split("::")` in `logstash/util` ? WDTY